doc/_scripts/gen_boards_catalog.py

```diff
@@ -78,6 +227,15 @@ def get_catalog():
     boards = list_boards.find_v2_boards(args_find_boards)
     systems = list_hardware.find_v2_systems(args_find_boards)
     board_catalog = {}
+    board_devicetrees = {}
+
+    if generate_hw_features:
+        logger.info("Running twister in cmake-only mode to get Devicetree files for all boards")
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            run_twister_cmake_only(tmp_dir)
+            board_devicetrees = gather_board_devicetrees(Path(tmp_dir))
```

Comment:
how long does this take? should turbo mode be ON by default here?

Another more fundamental question: is this information really useful in practice?

![image](https://github.com/user-attachments/assets/9ccaf732-08cf-4946-b6dc-716d84874d4b)

In the picture above, I do not care at all of more than 50% of the table entries. When looking at a board, I do usually care about external components available: QSPI flash? LEDSs? Sensors? Actual SoC (not what the SoC offers)?

Reply:
> How long does this take? Should turbo mode be ON by default here?

12 min or so, YMMV of course depending on your machine. I agree it's not fast, and we could likely have some cron produce a downloadable snapshot or something, I guess.

It is ON by default, i.e make html-fast turboes both the DT binding docs and this new feature.

> In the picture above, I do not care at all about more than 50% of the table entries. When looking at a board, I do usually care about external components available: QSPI flash? LEDSs? Sensors? Actual SoC (not what the SoC offers)?

I think that's what the search feature is for in the catalog "index". Did you try it? I.e. "I want a board with i2c+lora+display, what do you have in store for me Zephyr?"

Reply:
> > In the picture above, I do not care at all about more than 50% of the table entries. When looking at a board, I do usually care about external components available: QSPI flash? LEDSs? Sensors? Actual SoC (not what the SoC offers)?
> 
> I think that's what the search feature is for in the catalog "index". Did you try it? I.e. "I want a board with i2c+lora+display, what do you have in store for me Zephyr?"

Yes, seen it (not reviewed yet). What I mean is that, who cares about entries like "Clock controller", "Interrupt Controller", etc. or things like "NRF5 GPIOTE node", what does this really tell from a board usability perspective? When searching, this irrelevant information is probably less important as you write what you want to see.

Reply:
>  [...] entries like "Clock controller", "Interrupt Controller" [...]

I actually considered filtering those binding types as I agree they are of questionable value for most users. But my initial goal was to align with the way the table is currently populated for most boards, which, for better or worse, goes to that level of detail already ; so if anything, the new directive is just helping make these tables -- whether they're "flawed" or not -- more accurate.

> When searching, this irrelevant information is probably less important as you write what you want to see.

Most certainly, yep.

Reply:
Actually, there a lot of people relying on these tables (people seeking for level of support of some series or board and questioning "how come I2C is not supported ?" while the table is just not up to date).
And you can always try to educate people on how to understand Zephyr, you'll always get new comers asking the same question again and again because the initial reflex is to look at the board doc.

Alternative is to purely remove the tables. Having a list of boards highlights would end up in the same status as we have today, just with less entries in the list.

Reply:
I'll comment that different consumers have different needs. As an SOC vendor, the boards we upstream represent "kitchen sink" developer kits that are able to do many different things. But the primary objective is for us to showcase the capabilities of the SOC. We are not in the business of selling mass market boards... just the chips on those boards.

So an aspect we have historically struggled with is keeping the features up to date. Having a mechanism that automatically generates this is useful.